generator/generator.py

```diff
@@ -1,31 +1,67 @@
-#import numpy
+import numpy as np
 import sys
 import csv
 import random
+import matplotlib.pyplot as plt
+import scipy.stats
 
 def generator(shards, deltaT, N, mean, std):
-    tmp_list = []
     vectors = []
+    tasks_in_deltaT = []
+    requests = []
+    time_marks_in_deltaT = []
 
-    for x in range(shards):
-        for y in range(N):
-            tmp_list.append(random.randint(0, 10))
-        vectors.append(tmp_list)
-        tmp_list = []
-
+    for z in range(shards):        
+        y = np.random.normal(mean, std, N).tolist()
+        y = [round(num, 3) for num in y]
+        vectors.append(y)
+        y = []
 
     vectors_file = open("./generator/vectors.csv","w")
-    for x in vectors:
-        vector_string = str(vectors.index(x)+1) +","+ ','.join(map(str, x)) + "\n"
-        vectors_file.write(vector_string)
-
-
+    for z in vectors:
+       vector_string = str(vectors.index(z)+1) +","+ ','.join(map(str, z)) + "\n"
+       vectors_file.write(vector_string)
+    vectors_file.close()
+            
+    for n in range(N):
+        tmp = 0
+        for v in vectors:
+            if(v[n]>0):
+                tmp = tmp + 1
+        tasks_in_deltaT.append(tmp)
 
 
+    for n in range(N):
+        tmp_deltaT_start = n * deltaT
+        tmp_deltaT_end = (n + 1) * deltaT
+        time_marks = np.random.uniform(tmp_deltaT_start, tmp_deltaT_end, tasks_in_deltaT[n])
+        time_marks = [round(tm, 2) for tm in time_marks]
+        time_marks_in_deltaT.append(time_marks)
 
+    tmp_req = []
+    for n in range(N):
+        task_iterator = tasks_in_deltaT[n] - 1
+        for v in vectors:
+            if(v[n]>0):
```

Comment:
can it be false? when exactly?

Reply:
when standard deviation would be too big

Reply:
and also random can sometimes return 0